python-package/lightgbm/basic.py

```diff
@@ -2459,8 +2460,8 @@ def __init_from_pyarrow_table(
         ref_dataset: Optional[_DatasetHandle],
     ) -> "Dataset":
         """Initialize data from a PyArrow table."""
-        if not PYARROW_INSTALLED:
-            raise LightGBMError("Cannot init dataframe from Arrow without `pyarrow` installed.")
+        if not (PYARROW_INSTALLED and CFFI_INSTALLED):
+            raise LightGBMError("Cannot init dataframe from Arrow without `pyarrow` and `cffi` installed.")
```

Comment:
```suggestion
            raise LightGBMError("Cannot init Dataset from Arrow without `pyarrow` and `cffi` installed.")
```

This really should be `Dataset`, not `dataframe`... I'll make that change when I push testing changes.

Reply:
Fixed in https://github.com/microsoft/LightGBM/pull/6785/commits/e72d5e21521eb23492b689a3a361fa927286b3d3.

In that commit, I also removed backticks from these log messages, in favor of single quotes. Special characters in log messages can occasionally be problematic.

I know these things were already there before this PR, but might as well fix them right here while we're touching these lines.